File: server/src/main/java/com/example/server/model/Paper.java
package com.example.server.model;

import com.example.server.model.common.BaseEntity;
import org.babyfish.jimmer.sql.*;

import java.util.List;


@Entity
@Table(name = "paper")
public interface Paper extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    String title();

    Integer expired();

    @ManyToMany
    @JoinTable(name = "paper_question_mapping",
            joinColumnName = "paper_id",
            inverseJoinColumnName = "question_id")
    List<Question> questions();
}

File: website/src/__generated/model/entities/Paper.ts
import type { Question } from './';

export interface Paper {
    
    readonly createdTime: string;
    
    readonly modifiedTime: string;
    
    readonly id: number;
    
    readonly title: string;
    
    readonly expired?: number;
    
    readonly questions: ReadonlyArray<Question>;
    
    readonly question: number;
}
